examples/react-router/app/components/recipe_adder.jsx

```diff
@@ -0,0 +1,66 @@
+var React = require("react"),
+    Router = require("react-router"),
+    RouteHandler = Router.RouteHandler,
+    Link = Router.Link,
+    Navigation = Router.Navigation,
+    State = Router.State,
+    Fluxxor = require("../../../../");
+
+var RecipeForm = require("../forms/recipe_form.jsx");
+
+var RecipeAdder = React.createClass({
+  mixins: [
+    Fluxxor.FluxMixin(React),
+    State,
+    Navigation
+  ],
+
+  render: function() {
+    return this.renderWithLayout(
+      <div>
+        <form onSubmit={this.onSubmit}>
+          <RecipeForm ref="form" />
+          <input type="submit" value="Save" />
+        </form>
+      </div>
+    );
+  },
+
+  renderWithLayout: function(content) {
+    return (
+      <div>
+        {content}
+        <hr />
+        <Link to="home">Home</Link>
+        {" | "}<Link to="add-recipe">Add New Recipe</Link>
+      </div>
+    );
+  },
+
+  onSubmit: function(e) {
+    e.preventDefault();
+
+    var newRecipe = this.refs.form.getValue();
+    if (newRecipe) {
+      this.getFlux().actions.recipes.add(
+        newRecipe.name,
+        newRecipe.description,
+        newRecipe.ingredients,
+        newRecipe.directions,
+        function(recipe) {
```

Comment:
Application data flow should definitely be one way, but for side-effect-y things like this, I debate between strict adherence to the rule versus pragmatic breaking of the rule. In this case, I think I prefer that the component handles the redirect, but to do so it needs to know that the record has been created and what its ID is.

One alternative approach is to pass the router to the stores that need them and to transition them there; I don't *hate* this technique, and will consider updating the example.

```diff
diff --git a/examples/react-router/app/actions.jsx b/examples/react-router/app/actions.jsx
index e7722f1..c31ecda 100644
--- a/examples/react-router/app/actions.jsx
+++ b/examples/react-router/app/actions.jsx
@@ -6,13 +6,13 @@ var c = {
 
 var methods = {
   recipes: {
-    add: function(name, desc, ingredients, directions, callback) {
+    add: function(name, desc, ingredients, directions, transitionToRecipe) {
       this.dispatch(c.ADD_RECIPE, {
         name: name,
         description: desc,
         ingredients: ingredients,
         directions: directions,
-        callback: callback
+        transitionToRecipe: transitionToRecipe
       });
     },
 
diff --git a/examples/react-router/app/app.jsx b/examples/react-router/app/app.jsx
index 0ca3f4a..2480c10 100644
--- a/examples/react-router/app/app.jsx
+++ b/examples/react-router/app/app.jsx
@@ -8,8 +8,10 @@ var actions = require("./actions.jsx"),
 
 require("./style.less");
 
+var router = Router.create({routes: routes});
+
 var stores = {
-  recipe: new RecipeStore()
+  recipe: new RecipeStore({router: router})
 };
 
 var flux = new Fluxxor.Flux(stores, actions.methods);
@@ -28,7 +30,7 @@ flux.actions.recipes.add(
   "In a blender combine strawberries, milk, yogurt, sugar and vanilla. Toss in the ice. Blend until smooth and creamy. Pour into glasses and serve."
 );
 
-Router.run(routes, function(Handler) {
+router.run(function(Handler) {
   React.render(
     <Handler flux={flux} />,
     document.getElementById("app")
diff --git a/examples/react-router/app/components/recipe_adder.jsx b/examples/react-router/app/components/recipe_adder.jsx
index 7473b54..9c701c4 100644
--- a/examples/react-router/app/components/recipe_adder.jsx
+++ b/examples/react-router/app/components/recipe_adder.jsx
@@ -2,7 +2,6 @@ var React = require("react"),
     Router = require("react-router"),
     RouteHandler = Router.RouteHandler,
     Link = Router.Link,
-    Navigation = Router.Navigation,
     State = Router.State,
     Fluxxor = require("../../../../");
 
@@ -11,8 +10,7 @@ var RecipeForm = require("../forms/recipe_form.jsx");
 var RecipeAdder = React.createClass({
   mixins: [
     Fluxxor.FluxMixin(React),
-    State,
-    Navigation
+    State
   ],
 
   render: function() {
@@ -47,9 +45,7 @@ var RecipeAdder = React.createClass({
         newRecipe.description,
         newRecipe.ingredients,
         newRecipe.directions,
-        function(recipe) {
-          this.transitionTo("recipe", {id: recipe.id});
-        }.bind(this)
+        true
       );
     }
   },
diff --git a/examples/react-router/app/recipe_store.jsx b/examples/react-router/app/recipe_store.jsx
index 138071a..f913c07 100644
--- a/examples/react-router/app/recipe_store.jsx
+++ b/examples/react-router/app/recipe_store.jsx
@@ -5,7 +5,9 @@ var actions = require("./actions.jsx");
 var NOT_FOUND_TOKEN = {};
 
 var RecipeStore = Fluxxor.createStore({
-  initialize: function() {
+  initialize: function(options) {
+    this.router = options.router;
+
     this.recipeId = 0;
     this.recipes = {};
 
@@ -37,8 +39,8 @@ var RecipeStore = Fluxxor.createStore({
     this.recipes[recipe.id] = recipe;
     this.emit("change");
 
-    if (payload.callback) {
-      payload.callback(recipe);
+    if (payload.transitionToRecipe) {
+      this.router.transitionTo("recipe", {id: recipe.id});
     }
   },
```